tedge/src/cli/mqtt/cli.rs

```diff
@@ -37,7 +38,18 @@ pub enum TEdgeMqttCli {
 }
 
 impl BuildCommand for TEdgeMqttCli {
-    fn build_command(self, _context: BuildContext) -> Result<Box<dyn Command>, crate::ConfigError> {
+    fn build_command(self, context: BuildContext) -> Result<Box<dyn Command>, crate::ConfigError> {
+        let port = match context
+            .config_repository
+            .load()?
+            .query(MosquittoPortSetting)
+        {
+            Ok(p) => p,
+            _ => Port(DEFAULT_PORT),
+        };
```

Comment:
You don't need this match. You already set default in tedge_config library. If `mosquitto.port` is not set by user, `mosquitto.port` will use `1883`.
So, this can be simple and remove `DEFAULT_PORT`.
```suggestion
        let port = context
            .config_repository
            .load()?
            .query(MosquittoPortSetting)?;
```